src/librustc/middle/resolve_lifetime.rs

```diff
@@ -131,10 +131,10 @@ impl Region {
     fn shifted(self, amount: u32) -> Region {
```

Comment:
I think we could change this from `amount: u32` to `amount: usize` and avoid the casts below.

Reply:
(but see my comment below about maybe changing `shifted_in` instead)